### PR TITLE
AN-109441: Enhance ecoding support for byte & short

### PR DESCRIPTION
**Pull request by yznima**

### CODE REVIEW

**Comment by yznima** [comment]
@appian/datalayer-ha Could you review?

**Review by jkessler93 — COMMENTED** [review]
Can you add a description and link to the Travis CI?

File: decode_test.go [hunk 1]
```diff
@@ -132,7 +132,7 @@ func TestIntVector(t *testing.T) {
 
 }
 
-// `byte$til 5
+// 0x0001020304
```
Comment (jkessler93) [inline decode_test.go:135]: Can you include in the comment what this represents?

**Review by jkessler93 — APPROVED** [review]
Looks great!

File: .travis.yml [hunk 2]
```diff
@@ -3,6 +3,8 @@ language: go
 go:
   - 1.7
   - 1.8
+  - 1.9
+  - "1.10"
```
Comment (sbenderli) [inline .travis.yml:7]: why does this need to be a string

Comment (jkessler93) [reply .travis.yml:7]: Otherwise, it interprets it as 1.1

File: encode_test.go [hunk 3]
```diff
@@ -10,23 +10,42 @@ import (
...
-	{"1i", Int(1), IntBytes},
-	{"enlist 1i", &K{KI, NONE, []int32{1}}, IntVectorBytes},
-	{"`byte$til 5", &K{KG, NONE, []byte{0, 1, 2, 3, 4}}, ByteVectorBytes},
-	{"\"GOOG\"", &K{KC, NONE, "GOOG"}, CharArrayBytes},
+
+	// UUID
+	{"8c6b8b64-6815-6084-0a3e-178401251b68", &K{-UU, NONE, uuid.UUID{0x8c, 0x6b, 0x8b, 0x64, 0x68, 0x15, 0x60, 0x84, 0x0a, 0x3e, 0x17, 0x84, 0x01, 0x25, 0x1b, 0x68}}, GuidBytes},
+	{"00010203-0405-0607-0809-0a0b0c0d0e0f 10111213-1415-1617-1819-1a1b1c1d1e1f", &K{UU, NONE, []uuid.UUID{{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f}, {0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f}}}, GuidVecBytes},
+
+	// Byte/Int8
```
Comment (yznima) [inline encode_test.go:25]: Byte/Uint8

**Review by sbenderli — APPROVED** [review]
LGTM after in-person CR.

**Pull request merged**